instance/models/utils.py

```diff
@@ -651,3 +653,18 @@ def _is_json_serializable(obj):
         :return: Boolean True if object is list or dictionary, False otherwise.
         """
         return isinstance(obj, dict) or isinstance(obj, list) or isinstance(obj, bool)
+
+
+def retry_session(retries=5, session=None, backoff_factor=0.3, status_forcelist=(500, 502, 503, 504)):
+    session = session or requests.Session()
+    retry = Retry(
+        total=retries,
+        read=retries,
+        connect=retries,
+        backoff_factor=backoff_factor,
+        status_forcelist=status_forcelist,
+    )
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
```

Comment:
@viadanna, is there a reason we need HTTP?

Reply:
Not for Github, I can remove it but it'll make the feature less generic.